server/src/termdb.sql.js

```diff
@@ -51,14 +51,13 @@ return an array of sample names passing through the filter
 	const filter = await getFilterCTEs(qfilter, ds) // if qfilter is blank, it returns null
 	const sql = ds.cohort.db.connection.prepare(
 		filter
-			? `WITH ${filter.filters} SELECT sample as id, name FROM ${filter.CTEname} join sampleidmap on sample = sampleidmap.id`
-			: 'SELECT id, name FROM sampleidmap' // both statements must return sample id as a uniform behavior
+			? `WITH ${filter.filters} SELECT sample as id, name FROM ${filter.CTEname} join sampleidmap on sample = sampleidmap.id 
+			   where (sampleidmap.type = ${type} OR sampleidmap.type is NULL OR sampleidmap.type = '')`
```

Comment:
@airenzp I think there should be quotes around `${type}`

Reply:
thanks! I added the quoutes now. As for the null check, when the samples are imported from the sampleidmap file generated,  the type column if not provided is filled with empty values, not null. That is why I need to check this as well.